src/fs/metadata_log.cc

```diff
@@ -225,8 +235,11 @@ void metadata_log::memory_only_delete_dir_entry(inode_info::directory& dir, std:
 
 void metadata_log::schedule_flush_of_curr_cluster() {
     // Make writes concurrent (TODO: maybe serialized within *one* cluster would be faster?)
```

Comment:
Maybe
if (_read_only) { ... }
should be added here?

Reply:
I think it should also be added in schedule_background_task().

Reply:
It should also be in `schedule_flush_of_curr_cluster_and_change_it_to_new_one` and `append_ondisk_entry`.
IMO, checking for that in every operation is pointless and omission(i.e. bug)-prone. This check should be done in the closest places to writing to disk (or to disk buffer -- errors would come up earlier than flush_log()) so that no write to disk would happen. Since these functions are not noexcept and not asynchronous I think it is OK to just throw exceptions there, but somebody more experienced with such throwing should check that it is correct to do since there could be dangerous (see https://github.com/scylladb/seastar/blob/37a22d9de6b20ee3ecd76cbbec09c343836d7b72/doc/tutorial.md#exceptions-vs-exceptional-futures).